src/trie.rs

```diff
@@ -16,9 +16,20 @@ fn longest_common_prefix(s1: &str, s2: &str) -> usize {
     s1.chars().zip(s2.chars()).take_while(|t| t.0 == t.1).count()
 }
 
+fn first_char_unwrap(s: &String) -> char {
+    s.chars().nth(0).unwrap()
+}
+
 // taken from: http://stackoverflow.com/questions/28392008/more-concise-hashmap-initialization
 macro_rules! hashmap {
     ($( $key: expr => $val: expr ),*) => {{
+         let map = ::std::collections::HashMap::new();
```

Comment:
Same comment as before regarding the `vecdeque` macro and also the same solution (remove this macro and replace all instances of `hashmap![]` with `HashMap::new()`.